src/CalendarDate.ts

```diff
@@ -0,0 +1,171 @@
+  }
+
+  private static isLeapYear(year: number) {
+    return (year % 4 === 0 && year % 100 !== 0) || year % 400 === 0;
+  }
+
+  static parseString(isoString: string): CalendarDate {
+    const split = isoString.split('-');
+    if (split.length === 3) {
+      const year = parseInt(split[0]);
+      const month = parseInt(split[1]);
+      const day = parseInt(split[2]);
+      if (!isNaN(year) && !isNaN(month) && !isNaN(day)) {
+        return new CalendarDate(year, month - 1, day);
+      }
+    }
+    throw new Error(
+      `CalendarDate Validation Error: Input ${isoString.toString()} is not a valid ISO Date String.`,
+    );
+  }
+
+  /**
+   * Returns the ISO string representation yyyy-MM-dd
+   */
+  toString(): string {
+    return `${this.year.toString().padStart(4, '0')}-${(this.month + 1)
+      .toString()
+      .padStart(2, '0')}-${this.day.toString().padStart(2, '0')}`;
+  }
+
+  /**
+   * Used by JSON stringify method.
+   */
+  toJSON(): string {
+    return this.toString();
+  }
+
+  toDate(): Date {
+    return new Date(this.unixTimestampInSeconds * 1000);
+  }
+
+  /**
+   * Returns the unix timestamp in seconds.
+   */
+  valueOf(): number {
+    return this.unixTimestampInSeconds;
+  }
+
+  equals(calendarDate: CalendarDate): boolean {
+    return this.valueOf() === calendarDate.valueOf();
+  }
+
+  /**
+   * Returns a new CalendarDate with the specified amount of months added.
+   * If enforceEndOfMonth is set to true and the addition will cause an overflow to the next month, it will always return the last day of the month before.
+   */
+  addMonths(amount: number, enforceEndOfMonth = false): CalendarDate {
+    const date = this.toDate();
+    date.setMonth(this.month + amount);
+    if (enforceEndOfMonth && date.getMonth() !== (this.month + amount) % 12) {
+      date.setDate(0);
+    }
+    return new CalendarDate(date.getFullYear(), date.getMonth(), date.getDate());
+  }
+
+  /**
+   * Returns a new CalendarDate with the specified amount of days added.
+   * Allows overflow.
+   */
+  addDays(amount: number): CalendarDate {
+    const date = this.toDate();
+    date.setDate(this.day + amount);
+    return new CalendarDate(date.getFullYear(), date.getMonth(), date.getDate());
+  }
+
+  getLastDayOfMonth(): CalendarDate {
+    let maxDayOfMonth = daysInMonth[this.month];
+    if (this.month === 1 && CalendarDate.isLeapYear(this.year)) {
+      maxDayOfMonth = 29;
+    }
+    return new CalendarDate(this.year, this.month, maxDayOfMonth);
+  }
+
+  getFirstDayOfMonth(): CalendarDate {
+    return new CalendarDate(this.year, this.month, 1);
+  }
+
+  /**
+   * subtracts the input date from this date and returns the difference in days
+   */
+  getDifferenceInDays(calendarDate: CalendarDate, absolute?: boolean): number {
+    const days = (this.valueOf() - calendarDate.valueOf()) / DAY_IN_SECONDS;
+    return absolute ? Math.abs(days) : days;
+  }
+
+  isInsideOfDateRange(calendarDateRange: CalendarDateRange): boolean {
```

Comment:
i would keep calenderdate independet from CalenderDate and move it to CalenderDateRange and pass the date instead